model/operation.cc

```diff
@@ -435,16 +435,42 @@ namespace minsky
     switch (classify(type()))
       {
       case function: case reduction: case scan: case tensor:
-        if (check && !ports[1]->units(check).empty())
-          throw_error("function input not dimensionless");
-        return {};
+        switch (type())
+          {
+		  case percent:
+		    { 
+		     // Add % sign to units from input to % operator		
+             if (!ports[0]->wires().empty() && !ports[1]->wires().empty())  
+               {
+				 if (auto vV=dynamic_cast<VariableValue*>(&ports[0]->wires()[0]->to()->item())) 
+	             {    
+				   auto r=ports[1]->units(check);	 
+				   if (!r.empty()) {
```

Comment:
I don't think this test is required. If r.empty(), then I'm pretty sure r.str() will return the empty string, in which case the true and false parts of this conditional are identical.

Reply:
I've removed the if conditional in question.